hub-saml/src/test/java/uk/gov/ida/saml/hub/api/HubTransformersFactoryTest.java

```diff
@@ -47,57 +49,59 @@ public void setUp() {
 
     @Test
     public void getIdaAuthnRequestFromHubToStringTransformer_testDoesNotContainKeyInfo() throws Exception {
-        //Given
-        SignatureAlgorithm signatureAlgorithm = new SignatureRSASHA256();
-        DigestAlgorithm digestAlgorithm = new DigestSHA256();
         Function<IdaAuthnRequestFromHub, String> eidasTransformer = new HubTransformersFactory().getIdaAuthnRequestFromHubToStringTransformer(
-            getKeyStore(),
+            getKeyStore(true),
             signatureAlgorithm,
             digestAlgorithm
         );
         IdaAuthnRequestFromHub idaAuthnRequestFromHub = IdaAuthnRequestBuilder.anIdaAuthnRequest()
             .withLevelsOfAssurance(Collections.singletonList(AuthnContext.LEVEL_3))
             .buildFromHub();
 
-        //When
         String apply = eidasTransformer.apply(idaAuthnRequestFromHub);
 
-        //Then the Authn Request string is returned
         Assert.assertNotNull(apply);
 
-        //And the Authn Request does not contain a KeyInfo section for Verify UK
         AuthnRequest authnReq = stringtoOpenSamlObjectTransformer.apply(apply);
         Assert.assertNotNull(authnReq);
         Assert.assertNull("The Authn Request does not contain a KeyInfo section for Verify UK", authnReq.getSignature().getKeyInfo());
     }
 
     @Test
     public void getEidasAuthnRequestFromHubToStringTransformer_testContainsKeyInfo() throws Exception {
-        //Given
-        SignatureAlgorithm signatureAlgorithm = new SignatureRSASHA256();
-        DigestAlgorithm digestAlgorithm = new DigestSHA256();
         Function<EidasAuthnRequestFromHub, String> eidasTransformer = new HubTransformersFactory().getEidasAuthnRequestFromHubToStringTransformer(
-            getKeyStore(),
+            getKeyStore(true),
             signatureAlgorithm,
             digestAlgorithm
         );
         EidasAuthnRequestFromHub eidasAuthnRequestFromHub = EidasAuthnRequestBuilder.anEidasAuthnRequest()
             .withLevelsOfAssurance(Collections.singletonList(AuthnContext.LEVEL_2))
             .buildFromHub();
 
-        //When
         String apply = eidasTransformer.apply(eidasAuthnRequestFromHub);
 
-        //Then the Authn Request string is returned
         Assert.assertNotNull(apply);
 
-        //And the Authn Request contains a KeyInfo section for eIDAS
         AuthnRequest authnReq = stringtoOpenSamlObjectTransformer.apply(apply);
         Assert.assertNotNull(authnReq);
         Assert.assertNotNull("The Authn Request contains a KeyInfo section for eIDAS", authnReq.getSignature().getKeyInfo());
     }
 
-    private static IdaKeyStore getKeyStore() throws Base64DecodingException {
+    @Test (expected = SamlTransformationErrorException.class)
```

Comment:
Is it worth capturing and asserting the text of the exception? This is because there are multiple places can throw the same exception for different reasons... 🤔

Reply:
Yes, that's a good point. I'll make the change.